File: ChatGPT/2.Estructura de Datos Pila (Stack)/javascript/pila.js
class Stack {
    constructor() { this.items = []; }
    push(item) { this.items.push(item); }
    pop() { return this.items.length === 0 ? null : this.items.pop(); }
    isEmpty() { return this.items.length === 0; }
}

const readline = require('readline').createInterface({
    input: process.stdin, output: process.stdout
});
const s = new Stack();

function menu() {
    console.log("\nOpciones: 1) Push 2) Pop 3) Ver pila 4) Salir");
    readline.question("Elija opción: ", op => {
        if (op === "1") {
            readline.question("Valor a apilar: ", v => {
                s.push(v);
                console.log("Valor apilado.");
                menu();
            });
        } else if (op === "2") {
            console.log("Valor desapilado:", s.pop());
            menu();
        } else if (op === "3") {
            console.log("Contenido pila:", s.items);
            menu();
        } else if (op === "4") {
            readline.close();
        } else {
            menu();
        }
    });
}
menu();

File: ChatGPT/2.Estructura de Datos Pila (Stack)/golang/pila.go
package main
import (
    "bufio"
    "fmt"
    "os"
    "strconv"
	
)

type Stack struct{ items []int }
func (s *Stack) Push(item int) { s.items = append(s.items, item) }
func (s *Stack) Pop() (int, bool) {
    if len(s.items) == 0 { return 0, false }
    val := s.items[len(s.items)-1]
    s.items = s.items[:len(s.items)-1]
    return val, true
}
func (s *Stack) IsEmpty() bool { return len(s.items) == 0 }

func main() {
    s := Stack{}
    scanner := bufio.NewScanner(os.Stdin)
    for {
        fmt.Println("\nOpciones: 1) Push 2) Pop 3) Ver pila 4) Salir")
        scanner.Scan()
        op := scanner.Text()
        switch op {
        case "1":
            fmt.Print("Valor a apilar: ")
            scanner.Scan()
            v, _ := strconv.Atoi(scanner.Text())
            s.Push(v)
            fmt.Println("Valor apilado.")
        case "2":
            val, ok := s.Pop()
            if ok { fmt.Println("Valor desapilado:", val) } else { fmt.Println("Pila vacía.") }
        case "3":
            fmt.Println("Contenido pila:", s.items)
        case "4":
            return
        }
    }
}
